apps/api/src/domain/models/song/queries/song-queries-fetch-by-id.e2e.spec.ts

```diff
@@ -0,0 +1,391 @@
+                    },
+                    expectedStatus: HttpStatusCode.notFound,
+                });
+            });
+        });
+    });
+
+    describe(`when the user is an admin`, () => {
+        // ADMIN see all
+        const assertAdminResult = async (song: EventSourcedSongViewModel) => {
+            await assertQueryResult({
+                app,
+                endpoint: buildDetailEndpoint(song.id),
+                seedInitialState: async () => {
+                    await songQueryRepository.create(song);
+                },
+                expectedStatus: HttpStatusCode.ok,
+            });
+        };
+
+        describe(`when the user is authenticated as a project admin`, () => {
+            const testUserWithGroups = new CoscradUserWithGroups(
+                buildTestInstance(CoscradUser, {
+                    roles: [CoscradUserRole.projectAdmin],
+                }),
+                []
+            );
+
+            beforeAll(async () => {
+                await setItUp(testUserWithGroups);
+            });
+
+            describe(`when there is a song with the given ID`, () => {
+                describe(`when the song is public`, () => {
+                    const publicSong = buildTestInstance(EventSourcedSongViewModel, {
+                        isPublished: true,
+                        accessControlList: new AccessControlList(),
+                    });
+
+                    it(`should return the expected result`, async () => {
+                        await assertQueryResult({
+                            app,
+                            endpoint: buildDetailEndpoint(publicSong.id),
+                            expectedStatus: HttpStatusCode.ok,
+                            seedInitialState: async () => {
+                                await songQueryRepository.create(publicSong);
+                            },
+                            checkResponseBody: async (body) => {
+                                expect(body).toMatchSnapshot();
+                            },
+                        });
+                    });
+                });
+
+                describe(`when the song is private`, () => {
+                    describe(`when the user is in the query ACL`, () => {
+                        it(`should return the song`, async () => {
+                            await assertAdminResult(
+                                buildTestInstance(EventSourcedSongViewModel, {
+                                    isPublished: false,
+                                    accessControlList: new AccessControlList().allowUser(
+                                        testUserWithGroups.id
+                                    ),
+                                })
+                            );
+                        });
+                    });
+
+                    describe(`when the user is not the query ACL`, () => {
+                        it(`should **still** return the song`, async () => {
```

Comment:
This is interesting, I'd be curious to know more

Reply:
There's not much to know. The current rule is that a COSCRAD (super) admin or Project Admin has role-based access to all resources that takes precedence over ACL (row-level) based access.